### PR TITLE
Controller for Subscription Line Items

### PR DESCRIPTION
**Pull request by alexblackie**

### CODE REVIEW

File: app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      render json: @line_item.to_json
+    else
+      render json: @line_item.errors.to_json, status: 422
+    end
+  end
+
+  private
+
+  def line_item_params
+    params.require(:subscription_line_item).permit(
```
Comment (brendandeere) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: This is great for now. I have a PR making these values configurable. so we will likely want to eventually change this to Config.subscription_line_item_attrubutes - [:subscribable_id]

File: app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      :max_installments,
+      :interval,
+      :quantity
+    )
+  end
+
+  def load_line_item
+    @line_item = SolidusSubscriptions::LineItem.find(params[:id])
```
Comment (brendandeere) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: Should this be a `find!`? if it cant find the line item this will return a 500 for calling update on the nil class

Comment (brendandeere) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: Do we do any sort of validation that the user can update the subscription line item? might be another pr to add an ability. But I notice we are only sending a user api token instead of an order_token. 

Do all users always get api tokens or will stores need to start generating them for each user?

Comment (alexblackie) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: For authorization, there's a check of the `current_api_user` in the action itself, which I'm not stoked about but works.

Wouldn't a `find!` also return a 500? Or does Rails magic that to a 404?

Comment (brendandeere) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: I think the Spree Resource controller catching the record not found error.... 

https://github.com/solidusio/solidus/blob/master/api/app/controllers/spree/api/base_controller.rb#L27

File: app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb [hunk 3]
```diff
@@ -2,6 +2,9 @@ class SolidusSubscriptions::Api::V1::LineItemsController < Spree::Api::BaseContr
   before_filter :load_line_item, only: :update
 
   def update
+    return render json: {}, status: 400 unless @line_item.order.completed?
```
Comment (brendandeere) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: I dont think this is what we want. We will want to let users change some details about their subscription from the cart (interval, quantity, etc) the order wont be completed at that point. Even after the order is completed we may want to let users modify their subscriptions, but not as sure about the user stories around that

File: app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb [hunk 4]
```diff
@@ -2,6 +2,9 @@ class SolidusSubscriptions::Api::V1::LineItemsController < Spree::Api::BaseContr
...
+    return render json: {}, status: 404 unless @line_item.order.user == current_api_user
```
Comment (brendandeere) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: Do we prefer 404 over 401 if the users isn't authorized to edit the line item?

File: app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb (hunk 3 above)
Comment (alexblackie) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: Ah, I completely misread that point in the ticket; this is backwards. :no_mouth:

File: app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb (hunk 4 above)
Comment (alexblackie) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: It's common to use 404 instead of 403 to hide the fact resources exist (as just knowing something _exists_ can sometimes be a vulnerability). However, I find it very annoying and in this case the benefits may be dubious...

File: app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb (hunk 1 above)
Comment (brendandeere) [inline app/controllers/solidus_subscriptions/api/v1/line_items_controller.rb]: No to pile on, but this can now use `SolidusSubscriptions::Config.subscription_line_item_attributes`. might want to subtract the subscribable_id from the list though
